pkg/referrerstore/oras/oras.go

```diff
@@ -157,62 +169,108 @@ func (store *orasStore) ListReferrers(ctx context.Context, subjectReference comm
 	return referrerstore.ListReferrersResult{Referrers: referrers}, nil
 }
 
-func (store *orasStore) GetBlobContent(ctx context.Context, subjectReference common.Reference, digest digest.Digest) ([]byte, error) {
-	registryClient, err := store.createRegistryClient(ctx, subjectReference)
+func (store *orasStore) GetBlobContent(ctx context.Context, subjectReference common.Reference, digest digest.Digest, blobDesc ...oci.Descriptor) ([]byte, error) {
+	var err error
+	repository, err := store.createRepository(ctx, subjectReference)
 	if err != nil {
 		return nil, err
 	}
 
-	ref := fmt.Sprintf("%s@%s", subjectReference.Path, digest)
-	desc, err := oras.Copy(ctx, registryClient, ref, store.localCache, "")
+	// resolve blob descriptor if not provided
+	var resolvedBlobDesc oci.Descriptor
+	if len(blobDesc) > 0 {
+		resolvedBlobDesc = blobDesc[0]
+	} else {
+		ref := fmt.Sprintf("%s@%s", subjectReference.Path, digest)
+		resolvedBlobDesc, err = repository.Blobs().Resolve(ctx, ref)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// check if blob exists in local ORAS cache
+	isCached, err := store.localCache.Exists(ctx, resolvedBlobDesc)
 	if err != nil {
 		return nil, err
 	}
 
-	return store.getRawContentFromCache(ctx, desc)
+	if !isCached {
+		// fetch blob content from remote repository
+		rc, err := repository.Fetch(ctx, resolvedBlobDesc)
+		if err != nil {
+			return nil, err
+		}
+
+		// push fetched content to local ORAS cache
+		err = store.localCache.Push(ctx, resolvedBlobDesc, rc)
```

Comment:
What is the cache eviction strategy? Do we need to worry caching these blob bytes with no limits set?

Reply:
Currently there's no mechanism for evicting the blobs from the local store. This would probably involve setting some sort of map to cache blob digests to some expiry date and then manually evict based off of that. Should we make this an enhancement issue separately? cc: @susanshi